From 1b3fcbd3eead8f1f59f3e9472319d58ce6b52bb7 Mon Sep 17 00:00:00 2001
From: Luca Boccassi <luca.boccassi@gmail.com>
Date: Thu, 10 Nov 2016 22:49:41 +0000
Subject: [PATCH] Problem: linker fails looking for dladdr

Solution: search and add it via AC_CHECK_LIB when building with
libunwind, as the backtrace function uses dladdr. This problem
only appears on some distributions and with some compiler/toolchain
versions.
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 4fb2757..d548f9f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -650,6 +650,7 @@ PKG_CHECK_MODULES(LIBUNWIND, [libunwind],
         AC_DEFINE(HAVE_LIBUNWIND, 1, [The libunwind library is to be used])
         AC_SUBST([LIBUNWIND_CFLAGS])
         AC_SUBST([LIBUNWIND_LIBS])
+        AC_CHECK_LIB([dl], [dladdr])
     ],
     [
         AC_MSG_WARN([Cannot find libunwind])
